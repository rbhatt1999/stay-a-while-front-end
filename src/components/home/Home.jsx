/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { BsFillStarFill } from 'react-icons/bs';
import ReactPaginate from 'react-paginate';
import { fetchHomeStays } from './HomeSlice';
import style from './Home.module.css';

const Home = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const token = localStorage.getItem('token');
  const [currentPage, setCurrentPage] = useState(0);
  const [totalPages, setTotalPages] = useState(0);
  const itemsPerPage = 6;
  const startIndex = currentPage * itemsPerPage;
  const endIndex = startIndex + itemsPerPage;
  const homeStayList = useSelector((state) => state.home.listings);

  useEffect(() => {
    dispatch(fetchHomeStays(token));
  }, [dispatch, token]);
  const handlePageChange = (selectedPage) => {
    setCurrentPage(selectedPage.selected);
  };

  useEffect(() => {
    if (homeStayList) {
      setTotalPages(Math.ceil(homeStayList.length / itemsPerPage));
    }
  }, [homeStayList]);

  const errMsg = useSelector((state) => state.home.error);
  const subset = homeStayList.slice(startIndex, endIndex);

  const handleClick = (id) => {
    navigate(`/home/${id}`);
  };

  if (homeStayList) {
    return (
      <div className={style.container}>
        <div className={style.cardContainer}>
          {subset.map((homeStay) => (
            <div
              key={homeStay.id}
              onClick={() => handleClick(homeStay.id)}
              onKeyDown={() => {}}
              role="button"
              tabIndex={homeStay.id}
              className={style.card}
            >
              <div className="card h-100">
                <div className="card-body">
                  {homeStay.images.length > 0
                  && (
                  <img
                    className={style.image}
                    src={homeStay.images[0].url}
                    alt={homeStay.name}
                  />
                  )}
                  <div className="d-flex justify-content-between mt-3">
                    <h4 className="f_size">{homeStay.location}</h4>
                    <h6 className="">
                      <BsFillStarFill />
                      {homeStay.rating}
                    </h6>
                  </div>
                  <h5 className="hse_name">{homeStay.name}</h5>
                </div>
              </div>
            </div>
          ))}
        </div>
        <ReactPaginate
          itemClass="pageitem"
          linkClass="pagelink"
          previousLabel="Prev"
          nextLabel="Next"
          pageCount={totalPages}
          onPageChange={handlePageChange}
          containerClassName="pagination justify-content-center"
          pageClassName="pageitem"
          pageLinkClassName="pagelink"
          activeClassName="active"
          previousClassName="pageitem"
          previousLinkClassName="pagelink"
          nextClassName="pageitem"
          nextLinkClassName="pagelink"
        />
      </div>
    );
  }

  return errMsg;
};
export default Home;
