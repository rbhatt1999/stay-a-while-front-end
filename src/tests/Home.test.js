import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Home from '../components/home/Home';
import store from '../components/configureStore';

describe('Home', () => {
  it('should render Home component', async () => {
    const { findByText } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    expect(await findByText('Cottage by Maxwell')).toBeInTheDocument();
  });
  it('Show match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
