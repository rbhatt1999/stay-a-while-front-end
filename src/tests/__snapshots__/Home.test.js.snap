// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Show match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-5"
  >
    <div
      class="row row-cols-1 row-cols-sm-2 row-cols-md-3 row-cols-xlg-4 g-3"
    >
      <div
        class="col"
      >
        <div
          class="card h-100"
        >
          <div
            class="card-body"
          >
            <div
              class="carousel slide"
              id="carouselExampleCaptions-1"
            >
              <div
                class="carousel-indicators"
              >
                <button
                  aria-current="true"
                  aria-label="Slide 1"
                  class="active"
                  data-bs-slide-to="0"
                  data-bs-target="#carouselExampleCaptions-1"
                  type="button"
                />
                <button
                  aria-current=""
                  aria-label="Slide 2"
                  class=""
                  data-bs-slide-to="1"
                  data-bs-target="#carouselExampleCaptions-1"
                  type="button"
                />
                <button
                  aria-current=""
                  aria-label="Slide 3"
                  class=""
                  data-bs-slide-to="2"
                  data-bs-target="#carouselExampleCaptions-1"
                  type="button"
                />
                <button
                  aria-current=""
                  aria-label="Slide 4"
                  class=""
                  data-bs-slide-to="3"
                  data-bs-target="#carouselExampleCaptions-1"
                  type="button"
                />
                <button
                  aria-current=""
                  aria-label="Slide 5"
                  class=""
                  data-bs-slide-to="4"
                  data-bs-target="#carouselExampleCaptions-1"
                  type="button"
                />
              </div>
              <div
                class="carousel-inner"
              >
                <div
                  class="carousel-item img-holder active"
                >
                  <img
                    alt="Slide 1"
                    class="d-block w-100 img-tag"
                    src="https://images.pexels.com/photos/2102587/pexels-photo-2102587.jpeg?auto=compress&cs=tinysrgb&w=600"
                  />
                </div>
                <div
                  class="carousel-item img-holder "
                >
                  <img
                    alt="Slide 2"
                    class="d-block w-100 img-tag"
                    src="https://images.pexels.com/photos/1428348/pexels-photo-1428348.jpeg?auto=compress&cs=tinysrgb&w=600"
                  />
                </div>
                <div
                  class="carousel-item img-holder "
                >
                  <img
                    alt="Slide 3"
                    class="d-block w-100 img-tag"
                    src="https://images.pexels.com/photos/2179214/pexels-photo-2179214.jpeg?auto=compress&cs=tinysrgb&w=600"
                  />
                </div>
                <div
                  class="carousel-item img-holder "
                >
                  <img
                    alt="Slide 4"
                    class="d-block w-100 img-tag"
                    src="https://images.pexels.com/photos/2346091/pexels-photo-2346091.jpeg?auto=compress&cs=tinysrgb&w=600"
                  />
                </div>
                <div
                  class="carousel-item img-holder "
                >
                  <img
                    alt="Slide 5"
                    class="d-block w-100 img-tag"
                    src="https://images.pexels.com/photos/1571450/pexels-photo-1571450.jpeg?auto=compress&cs=tinysrgb&w=600"
                  />
                </div>
              </div>
              <button
                class="carousel-control-prev"
                data-bs-slide="prev"
                data-bs-target="#carouselExampleCaptions-1"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="carousel-control-prev-icon"
                />
                <span
                  class="visually-hidden"
                >
                  Previous
                </span>
              </button>
              <button
                class="carousel-control-next"
                data-bs-slide="next"
                data-bs-target="#carouselExampleCaptions-1"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="carousel-control-next-icon"
                />
                <span
                  class="visually-hidden"
                >
                  Next
                </span>
              </button>
            </div>
            <div
              class="d-flex justify-content-between mt-3"
            >
              <h4
                class="f_size"
              >
                Sao Paulo, Brazil
              </h4>
              <h6
                class=""
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                  />
                </svg>
                3.0
              </h6>
            </div>
            <h5
              class="hse_name fs-9"
            >
              Cottage by Maxwell
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
